src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,124 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        T value;
+        Node<T> next;
+        Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.next = next;
+            this.value = element;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> temp = new Node<>(tail, value, null);
+        if (head == null) {
+            head = temp;
+            tail = head;
+        } else {
+            tail.next = temp;
+            tail = temp;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
 
+        Node<T> oldNode = getNodeByIndex(index);
+        Node<T> newNode = new Node<T>(oldNode.prev, value, oldNode);
+
+        if (oldNode.prev != null) {
+            oldNode.prev.next = newNode;
+            oldNode.prev = newNode;
+        } else {
+            head = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> temp = getNodeByIndex(index);
+        T oldValue = temp.value;
+        temp.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> temp = getNodeByIndex(index);
+        if (temp.prev != null) {
+            temp.prev.next = temp.next;
+        } else {
+            head = temp.next;
+        }
+        if (temp.next != null) {
+            temp.next.prev = temp.prev;
+        } else {
+            tail = temp.prev;
+        }
+        size--;
+        return temp.value;
```

Comment:
let's move this logic to the separate method delete()/unlink() and use it here and in the method remove(T object)